src/cefpython.pyx

```diff
@@ -1017,11 +1017,12 @@ def Shutdown():
     # and closes browser automatically if you give it some time.
     # If the time was not enough, then there is an emergency plan,
     # the code block further down that checks len(g_pyBrowsers).
-    for _ in range(20):
-        for __ in range(10):
-            with nogil:
-                CefDoMessageLoopWork()
-        time.sleep(0.01)
+    if not g_applicationSettings["multi_threaded_message_loop"]:
```

Comment:
"multi_threaded_message_loop" key may not always exist in g_applicationSettings. You should use the GetAppSetting() function instead. Same for the other case later in code.